aip/general/0236.md

```diff
@@ -0,0 +1,351 @@
+  methods.
+- The first time the "preview" custom method is used, the system **must** create
+  `preview_metadata` and do the following: 
+  - It **must** set the `state` to `ACTIVE`
+  - It **must** populate `start_time` with the current time. 
+    - `start_time` **must** be updated every time the status is changed to
+      `ACTIVE`.
+  - It **must** set a `log_prefix` string, which is a predefined constant
+    hard coded by the system developers. 
+    - The same value is used for previewing experiments for the given resource
+      resource type. For example, "FirewallPolicyPreviewLog" for FirewallPolicy.
+- When the "stop" custom method is used, the system **must** do the following:
+  - It **must** set the `state` to `INACTIVE`
+  - It **must** populate the `stop_time` with the current time. 
+
+### Methods
+
+#### create
+
+- The resource **must** be created using long-running 
+  [Create][aip-133-long-running] and 
+  `google.longrunning.operation_info.response_type` **must** be
+  PolicyExperiment.
+- Creating a new experiment to preview **must** support the following use
+cases:
+  - Preview a new policy.
+  - Preview an update to an already live policy.
+  - Preview a deletion of a current policy.
+- For the update and delete use cases, the `policy` field in the experiment
+  **must** have the full payload of the live policy copied into it, including
+  the name. 
+  - The user **must** set the rules to the new intended state to preview an
+    update.
+  - The user **must** set set the rules to represent a no-op to preview a
+    delete.
+- To preview a new policy, the system must do the following: 
+  - If the system does not support a nested collection without a live policy, 
+    the user **must** create a live policy and set the rules to represent a
+    no-op. For example, the rules of a no-op policy **may** be empty.
+    - An experiment is created as a child of the no-op policy.
+- If the system supports previewing multiple experiments for a live policy,
+  calling "create" more than once **must** create multiple experiments.
+
+#### update
+
+- The resource **must** be updated using long-running 
+  [Update][aip-134-long-running] and 
+  `google.longrunning.operation_info.response_type` **must** be
+  PolicyExperiment.
+- The name inside `policy` **must not** change but the other fields can in
+  order to change the experiment being previewed.
+- The system **must** set the `state` to INACTIVE if the `state` was ACTIVE at
+  the time of an update.
+  - This is so the user can easily distinguish between different versions of
+    the experiment being previewed.
+
+#### get
+- The standard method, [Get][aip-131], **must** be included for PolicyExperiment
+  resource types.
+
+#### list
+
+- The standard method, [List][aip-132], **must** be included for
+  PolicyExperiment resource types.
+- Filtering on `PreviewMetadata` indicates which experiments are actively
+  previewed.
```

Comment:
I think the filtering needs to be explained with an example. What would the string contents of the field, `filter` on the list request message be? For presence? For absence? Can you filter on a specific field in the metadata, like started after?